sys/kern/pci.c

```diff
@@ -35,18 +35,36 @@ static bool pci_device_present(device_t *pcid) {
 }
 
 static int pci_device_nfunctions(device_t *pcid) {
+  /* If first function of a device is invalid, then
+   * no more functions are present. */
   if (!pci_device_present(pcid))
     return 0;
   uint32_t hdrtype = pci_read_config(pcid, PCIR_HEADERTYPE, 1);
   return (hdrtype & PCIH_HDR_MF) ? PCI_FUN_MAX_NUM : 1;
 }
 
-static uint32_t pci_bar_size(device_t *pcid, int bar, uint32_t addr) {
-  pci_write_config(pcid, PCIR_BAR(bar), 4, -1U);
-  uint32_t res = pci_read_config(pcid, PCIR_BAR(bar), 4);
+static uint32_t pci_bar_size(device_t *pcid, int bar, uint32_t *addr) {
+  /* Memory and I/O space acceses must be disabled via the
```

Comment:
Please spell check your comments (this is not the first time I making that remark).

Reply:
Done.